src/python/pants/java/nailgun_protocol.py

```diff
@@ -245,7 +245,7 @@ def isatty_to_env(cls, stdin, stdout, stderr):
     def gen_env_vars():
       for fd_id, fd in zip(STDIO_DESCRIPTORS, (stdin, stdout, stderr)):
         is_atty = fd.isatty()
-        yield (cls.TTY_ENV_TMPL.format(fd_id), bytes([int(is_atty)]))
+        yield (cls.TTY_ENV_TMPL.format(fd_id), six.binary_type(str(int(is_atty))))
```

Comment:
You can also use `future.utils.binary_type()`. They're equivalent in meaning, so this isn't worth re-pushing, but as an FYI.

This, however, is not Py3 compatible! The call to `str()` will have different behavior on Py2 and Py3, because it's missing the import `from builtins import str`. Right now, in Py2 this code is translating to `bytes(bytes(int(is_atty)))`. In Py3, it translates to `bytes(unicode(int(is_atty)))`

To fix this, add `from builtins import str` and see if the tests pass.

Reply:
Ah! I had tried `from builtins import str`, but not after switching `bytes` to `six.binary_type` -- see [my comment above](https://github.com/pantsbuild/pants/pull/6181#discussion_r203564668).

While this doesn't *need* to block getting this fix in, considering that the fix was because of a change we made for python 3 compat, I would strongly prefer to have the fix you described so we don't have knowingly broken code out, so I just pushed dfd1be1 and will wait for CI to pass.

Reply:
may have to take a look at that linter soon then........

Reply:
I only have 2.5 weeks left at Foursquare so won’t be able to get to the linter anytime soon, unfortunately. 

I’d definitely be interested in helping after I finish the internship though! I’m planning to stay engaged with this Python 3 port the next year - can’t wait to replace every `str.format()` with f strings once we drop Python 2 :)